utils/frame/try-runtime/cli/src/lib.rs

```diff
@@ -328,10 +340,116 @@ where
 	Ok(())
 }
 
+async fn execute_block<Block, ExecDispatch>(
+	shared: SharedParams,
+	command: ExecuteBlockCmd,
+	config: Configuration,
+)-> sc_cli::Result<()>
+where
+	Block: BlockT + serde::de::DeserializeOwned,
+	Block::Hash: FromStr,
+	<Block::Hash as FromStr>::Err: Debug,
+	NumberFor<Block>: FromStr,
+	<NumberFor<Block> as FromStr>::Err: Debug,
+	ExecDispatch: NativeExecutionDispatch + 'static,
+{
+	let wasm_method = shared.wasm_method;
+	let execution = shared.execution;
+	let heap_pages = if shared.heap_pages.is_some() {
+		shared.heap_pages
+	} else {
+		config.default_heap_pages
+	};
+
+	let mut changes = Default::default();
+	let max_runtime_instances = config.max_runtime_instances;
+	let executor = NativeExecutor::<ExecDispatch>::new(
+		wasm_method.into(),
+		heap_pages,
+		max_runtime_instances,
+	);
+
+	let (mode, block) = match command.state {
+		State::Snap{ snapshot_path, url } => {
+			// `--header-at` is a required option for `snap`
+			let header_hash: Block::Hash = command.header_at
+				.expect("`--header-at` is a required option for `execute-block snap`.")
```

Comment:
Pretty sure it's possible to express this requirement in a way that can cause structopt/clap to enforce it for you with a nicer error message than an actual panic dump. See [`required_if`](https://docs.rs/clap/2.33.3/clap/struct.Arg.html#method.required_if).